Libs/MRML/Logic/vtkMRMLColorLogic.h

```diff
@@ -125,6 +176,118 @@ class VTK_MRML_LOGIC_EXPORT vtkMRMLColorLogic : public vtkMRMLAbstractLogic
+  /// Utility method that returns a colon separated triplet for the
+  /// SegmentedPropertyTypeModifier for this label in this color node.
+  /// Returns an empty string if all elements not found or defined, otherwise a
+  /// color separated tuple of the format CodeValue:CodingSchemeDesignator:CodeMeaning
+  std::string GetSegmentedPropertyTypeModifier(int label, const char *lutName = "GenericAnatomyColors");
+
+  /// Utility methods to look up the terminology for the AnatomicRegion
+  /// for a specific label in a color node.
+  /// Returns an empty string if not found or defined.
+  std::string GetAnatomicRegionCodeValue(int label, const char *lutName = "GenericAnatomyColors");
+  std::string GetAnatomicRegionCodeMeaning(int label, const char *lutName = "GenericAnatomyColors");
+  std::string GetAnatomicRegionCodingSchemeDesignator(int label, const char *lutName = "GenericAnatomyColors");
+  /// Utility method that returns a colon separated triplet for the
+  /// AnatomicRegion for this label in this color node.
+  /// Returns an empty string if all elements not found or defined, otherwise a
+  /// color separated tuple of the format CodeValue:CodingSchemeDesignator:CodeMeaning
+  std::string GetAnatomicRegion(int label, const char *lutName = "GenericAnatomyColors");
+
+  /// Utility methods to look up the terminology for the AnatomicRegionModifier
+  /// for a specific label in a color node.
+  /// Returns an empty string if not found or defined.
+  std::string GetAnatomicRegionModifierCodeValue(int label, const char *lutName = "GenericAnatomyColors");
+  std::string GetAnatomicRegionModifierCodeMeaning(int label, const char *lutName = "GenericAnatomyColors");
+  std::string GetAnatomicRegionModifierCodingSchemeDesignator(int label, const char *lutName = "GenericAnatomyColors");
+  /// Utility method that returns a colon separated triplet for the
+  /// AnatomicRegionModifier for this label in this color node.
+  /// Returns an empty string if all elements not found or defined, otherwise a
+  /// color separated tuple of the format CodeValue:CodingSchemeDesignator:CodeMeaning
+  std::string GetAnatomicRegionModifier(int label, const char *lutName = "GenericAnatomyColors");
+
+  /// Create a new empty terminology for this LUT and try to associate it with a color
+  /// node of the same name.
+  /// Returns true on success, false if lutName is empty or can't associate the new
+  /// terminology with a color node
+  /// \sa AssociateTerminologyWithColorNode
+  bool CreateNewTerminology(std::string lutName);
+  /// Return true if a terminology mapping already exists for this lutName,
+  // false if empty name or none found.
+  bool TerminologyExists(std::string lutName);
+  /// Link the color node with the terminology
+  /// Returns true on success, false if empty lut name, unable to find the color node
+  /// \sa CreateNewTerminology
+  bool AssociateTerminologyWithColorNode(std::string lutName);
```

Comment:
Can method `AssociateTerminologyWithColorNode` be protected ? 

This is an implementation details of `CreateNewTerminology`